src/System.Windows.Forms/tests/UnitTests/System/Resources/ResXResourceSetTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace System.Resources.Tests
+{
+    // NB: doesn't require thread affinity
+    public class ResXResourceSetTests : IClassFixture<ThreadExceptionFixture>
+    {
+        [Theory]
+        [InlineData("TestResources.resx", "SomeMissingTest1", null)]
+        [InlineData("TestResources.resx", "SomeMetadataTest1", "Some text for MetaData 1 node")]
+        [InlineData("TestResources.resx", "SomeMetadataTest2", "Some text for MetaData 2 node")]
+        [InlineData("TestResources.resx", "SomeDataTest1", "Some text for Data 1 node")]
+        [InlineData("TestResources.resx", "SomeDataTest2", "Some text for Data 2 node")]
+        [InlineData("TestResources.resx", "text.ansi", "Text")]
+        [InlineData("TestResources.resx", "text.utf8", "Привет")]
+        private void ResXResourceSet_TestFile(string resxFileName, string resourceName, string expected)
+        {
+            Assert.True(File.Exists(resxFileName), $@"RESX file ""{resxFileName}"" not found, make sure it's in the root folder of the unit test project");
+
+            using (ResXResourceSet resxSet = new ResXResourceSet(resxFileName))
+            {
+                Assert.NotNull(resxSet);
+
+                string strResXValue = resxSet.GetString(resourceName);
+
+                if (expected == null)
+                {
+                    Assert.Null(strResXValue);
+                }
+                else
+                {
+                    Assert.NotNull(strResXValue);
+                    Assert.Equal(expected, strResXValue);
+                }
```

Comment:
We can simplify it further:
```suggestion
                Assert.Equal(expected, strResXValue);
```

If `expected is null` the above assertion is the same as `Assert.Null(strResXValue)`; otherwise it assert both that `strResXValue is not null` and `strResXValue == expected`.

In general a unit test should have a cyclomatic complexity of 1 (i.e. no conditional branching) - otherwise you need to write a unit test for a unit test 🤣